pkg/cmd/browse/browse.go

```diff
@@ -0,0 +1,182 @@
+		`),
+		Annotations: map[string]string{
+			"IsCore": "true",
+			"help:arguments": heredoc.Doc(`
+				A browser location can be specified using arguments in the following format:
+				- by number for issue or pull request, e.g. "123"; or
+				- by path for opening folders and files, e.g. "cmd/gh/main.go"
+			`),
+			"help:environment": heredoc.Doc(`
+				To configure a web browser other than the default, use the BROWSER environment variable.
+			`),
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.BaseRepo = f.BaseRepo
+
+			if len(args) > 0 {
+				opts.SelectorArg = args[0]
+			}
+
+			if err := cmdutil.MutuallyExclusive(
+				"specify only one of `--branch`, `--projects`, `--wiki`, or `--settings`",
+				opts.Branch != "",
+				opts.WikiFlag,
+				opts.SettingsFlag,
+				opts.ProjectsFlag,
+			); err != nil {
+				return err
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runBrowse(opts)
+		},
+	}
+
+	cmdutil.EnableRepoOverride(cmd, f)
+	cmd.Flags().BoolVarP(&opts.ProjectsFlag, "projects", "p", false, "Open repository projects")
+	cmd.Flags().BoolVarP(&opts.WikiFlag, "wiki", "w", false, "Open repository wiki")
+	cmd.Flags().BoolVarP(&opts.SettingsFlag, "settings", "s", false, "Open repository settings")
+	cmd.Flags().StringVarP(&opts.Branch, "branch", "b", "", "Select another branch by passing in the branch name")
+
+	return cmd
+}
+
+func runBrowse(opts *BrowseOptions) error {
+	baseRepo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("unable to determine base repository: %w\nUse 'gh browse --help' for more information about browse\n", err)
+	}
+
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("unable to create an http client: %w\nUse 'gh browse --help' for more information about browse\n", err)
+	}
+	url := ghrepo.GenerateRepoURL(baseRepo, "")
+
+	if opts.SelectorArg == "" {
+		if opts.ProjectsFlag {
+			url += "/projects"
+		}
+		if opts.SettingsFlag {
+			url += "/settings"
+		}
+		if opts.WikiFlag {
+			url += "/wiki"
+		}
+		if opts.Branch != "" {
+			url += "/tree/" + opts.Branch + "/"
+		}
+	} else {
+		if isNumber(opts.SelectorArg) {
+			url += "/issues/" + opts.SelectorArg
+		} else {
+			arr, err := parseFileArg(opts.SelectorArg)
+			if err != nil {
+				return err
+			}
+			if opts.Branch != "" {
+				url += "/tree/" + opts.Branch + "/"
+			} else {
+				apiClient := api.NewClientFromHTTP(httpClient)
+				branchName, err := api.RepoDefaultBranch(apiClient, baseRepo)
+				if err != nil {
+					return err
+				}
+				url += "/tree/" + branchName + "/"
+			}
+			if opts.SelectorArg != "" {
+				if len(arr) > 1 {
+					url += arr[0] + "#L" + arr[1]
+				} else {
+					url += arr[0]
+				}
+			}
+		}
+	}
+
+	err = opts.Browser.Browse(url)
+	if opts.IO.IsStdoutTTY() && err == nil {
+		fmt.Fprintf(opts.IO.Out, "now opening %s in browser\n", url)
+	}
+
+	return err
+}
+
+func parseFileArg(fileArg string) ([]string, error) {
```

Comment:
Nitpick: the purpose of this function is to extract file path and line number from a string argument, but it returns a slice of strings as a result. This leaves slightly more responsibility to the caller, which now has to always check `len(result)` before using the result to see if a number was extracted.

To make it easier for the caller, could this function do something more useful? An idea that comes to my mind is: since the caller ultimately wants to convert the `<file>:<line>` argument into `<file>#L<line>` to prepare it for the URL, why not have this function do that preparation? In other words, this function can accept a string as input (representing the file argument from the command line) and have a string as output (representing the file as it appears in the URL, together with the optional `#L<line>` fragment).

Do this change only if you have time for polish!